charts/dns/templates/external-dns-deployment.yaml

```diff
@@ -5,45 +5,112 @@ metadata:
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "stateless-dns.serviceAccountName" . }}
+      {{- with .Values.shareProcessNamespace }}
+      shareProcessNamespace: {{ . }}
+      {{- end }}
+      {{- with .Values.podSecurityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.priorityClassName }}
+      priorityClassName: {{ . | quote }}
+      {{- end }}
+      {{- with .Values.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{- end }}
       containers:
         - name: external-dns
-          image: "{{ .Values.externalDNS.repository }}:{{ .Values.externalDNS.tag }}"
+          image: "{{ .Values.externalDNS.registry }}/{{ .Values.externalDNS.repository }}:{{ .Values.externalDNS.tag }}"
           imagePullPolicy: {{ .Values.externalDNS.pullPolicy }}
-          args:
-          - --source=ingress
-          - --source=service
-        {{- if default false .Values.externalDNS.sourceIstioObjects }}
-          - --source=istio-gateway
-          - --source=istio-virtualservice
-        {{- end }}
-          {{- range $zonename, $zonefile := .Values.pdns.zones }}
-          - --domain-filter={{ $zonename }}
-          {{- end }}
-          envFrom:
-          - configMapRef:
-              name: {{ include "externalDNS.fullname" . }}
+          {{- with .Values.externalDNS.securityContext }}
           securityContext:
-            runAsNonRoot: true
-            runAsUser: 65534
-            readOnlyRootFilesystem: true
-            capabilities:
-              drop: ["ALL"]
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          args:
+            - --log-level={{ .Values.externalDNS.logLevel }}
+            - --log-format={{ .Values.externalDNS.logFormat }}
+
+            - --interval={{ .Values.externalDNS.interval }}
+            {{- if .Values.externalDNS.triggerLoopOnEvent }}
+            - --events
+            {{- end }}
+
+            {{- range .Values.externalDNS.sources }}
+            - --source={{ . }}
+            {{- end }}
+            {{- range $zonename, $zonefile := .Values.zones }}
+            - --domain-filter={{ $zonename }}
+            {{- end }}
+            - --managed-record-types={{ .Values.externalDNS.managedRecordTypes }}
+ 
+            - --registry=txt
+            {{- if .Values.externalDNS.txtOwnerId }}
+            - --txt-owner-id={{ .Values.externalDNS.txtOwnerId }}
+            {{- end }}
+            {{- if .Values.externalDNS.txtPrefix }}
+            - --txt-prefix={{ .Values.externalDNS.txtPrefix }}
+            {{- end }}
+            {{- if .Values.externalDNS.txtSuffix }}
+            - --txt-suffix={{ .Values.externalDNS.txtSuffix }}
+            {{- end }}
+            {{- if .Values.externalDNS.txtWildcardReplacement }}
+            - --txt-wildcard-replacement={{ .Values.externalDNS.txtWildcardReplacement }}
+            {{- end }}
```

Comment:
I thought about it but I think I read it better this way.

To be able to `range` I should also move everything from `.Values.externalDNS.*` to `.Values.externalDNS.config.*` so I do not mix config with (for example) image ir securityContext values.

I have mixed feelings. What would you prefer?

Reply:
If we do not foresee more txt options, I'll leave it as it is. dpehaps using `with` just to avoid possible copypaste errors that are completely invisible to humans.